src/table.rs

```diff
@@ -0,0 +1,320 @@
+//! FATs managment.
+
+use super::filesystem::FatFileSystem;
+use super::utils::FileSystemIterator;
+use super::Cluster;
+use super::FatError;
+use super::FatFileSystemResult;
+use super::FatFsType;
+use byteorder::{ByteOrder, LittleEndian};
+use storage_device::StorageDevice;
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+/// Represent a cluster chan value.
+pub enum FatValue {
+    /// Represent a free cluster.
+    Free,
+
+    /// Represent a used cluster.
+    Data(u32),
+
+    /// Represent a corrupted cluster (bad sectors)
+    Bad,
+
+    /// Represent the end of a cluster chain.
+    EndOfChain,
+}
+
+/// Util iterator used to simplify iteration over cluster.
+pub struct FatClusterIter {
+    /// The last cluster returned.
+    current_cluster: Option<Cluster>,
+
+    /// The last FatValue used.
+    last_fat: Option<FatValue>,
+}
+
+impl FatClusterIter {
+    /// Create a new Cluster iteractor starting at ``cluster``.
```

Comment:
```suggestion
    /// Create a new Cluster iterator starting at ``cluster``.
```

You should set up a spell checker in your IDE :p